test/hotspot/jtreg/compiler/c2/irTests/UDivLNodeIdealizationTests.java

```diff
@@ -0,0 +1,161 @@
+    public void runMethod() {
+        long a = RunInfo.getRandom().nextLong();
+        a = (a == 0) ? 1 : a;
+        long b = RunInfo.getRandom().nextLong();
+        b = (b == 0) ? 1 : b;
+
+        long min = Long.MIN_VALUE;
+        long max = Long.MAX_VALUE;
+
+        assertResult(0, 0, true);
+        assertResult(a, b, false);
+        assertResult(min, min, false);
+        assertResult(max, max, false);
+    }
+
+    @DontCompile
+    public void assertResult(long a, long b, boolean shouldThrow) {
+        try {
+            Asserts.assertEQ(Long.divideUnsigned(a, a), constant(a));
+            Asserts.assertFalse(shouldThrow, "Expected an exception to be thrown.");
+        } catch (ArithmeticException e) {
+            Asserts.assertTrue(shouldThrow, "Did not expect an exception to be thrown.");
+        }
+
+        try {
+            Asserts.assertEQ(Long.divideUnsigned((a * b), b), retainDenominator(a, b));
+            Asserts.assertFalse(shouldThrow, "Expected an exception to be thrown.");
+        } catch (ArithmeticException e) {
+            Asserts.assertTrue(shouldThrow, "Did not expect an exception to be thrown.");
+        }
+
+        try {
+            Asserts.assertEQ(Long.divideUnsigned(a, Long.divideUnsigned(b, b)), identityThird(a, b));
+            Asserts.assertFalse(shouldThrow, "Expected an exception to be thrown.");
+        } catch (ArithmeticException e) {
+            Asserts.assertTrue(shouldThrow, "Did not expect an exception to be thrown.");
+        }
+
+        Asserts.assertEQ(Long.divideUnsigned(a, 1), identity(a));
+        Asserts.assertEQ(Long.divideUnsigned(a, Long.divideUnsigned(13, 13)), identityAgain(a));
+        Asserts.assertEQ(Long.divideUnsigned(a, 8), divByPow2(a));
+        Asserts.assertEQ(Long.divideUnsigned(a, Long.MIN_VALUE), divByPow2Big(a));
+        Asserts.assertEQ(Long.divideUnsigned(a, RANDOM_POWER_OF_2), divByPow2Random(a));
+        Asserts.assertEQ(Long.divideUnsigned(a, Long.divideUnsigned(-9214294468834361176L, -9214294468834361176L)), identityAgainButBig(a));
+    }
+
+    @Test
+    @IR(failOn = {IRNode.UDIV})
+    @IR(counts = {IRNode.DIV_BY_ZERO_TRAP, "1"})
+    // Checks x / x => 1
+    public long constant(long x) {
+        return Long.divideUnsigned(x, x);
+    }
+
+    @Test
+    @IR(failOn = {IRNode.UDIV})
+    // Checks x / 1 => x
+    public long identity(long x) {
+        return Long.divideUnsigned(x, 1L);
+    }
+
+    @Test
+    @IR(failOn = {IRNode.UDIV})
+    // Checks x / (c / c) => x
+    public long identityAgain(long x) {
+        return Long.divideUnsigned(x, Long.divideUnsigned(13L, 13L));
+    }
+
+    @Test
+    @IR(failOn = {IRNode.UDIV})
+    // Checks x / (c / c) => x
+    public long identityAgainButBig(long x) {
+        return Long.divideUnsigned(x, Long.divideUnsigned(-9214294468834361176L, -9214294468834361176L));  // Long.parseUnsignedLong("9232449604875190440") = -9214294468834361176L
```

Comment:
What is the comment for here?

Reply:
Is this `Long.MIN_VALUE`? Can you explain more in the comment, and move it on a separate line?